static/js/fenced_code.js

```diff
@@ -185,4 +185,20 @@ exports.process_fenced_code = function (content) {
     return output.join('\n');
 };
 
+const fence_length_re = /^ {0,3}(`{3,})/g;
+exports.get_unused_fence = (content) => {
+    // we only return ``` fences, not ~~~.
+    let current_max = 2;  // Start with two so we default to returning fence of length 3.
+    content.split('\n').forEach(line => {
+        const m = line.match(fence_length_re);
+        if (m) {
+            const fence = m[0].trim();
+            if (fence.length > current_max) {
+                current_max = fence.length;
+            }
+        }
+    });
+    return '`'.repeat(current_max + 1);
+};
```

Comment:
Using a multiline regex is simpler:

```js
const fence_length_re = /^ {0,3}(`{3,})/gm;
exports.get_unused_fence = (content) => {
    // we only return ``` fences, not ~~~.
    let length = 3;
    let match;
    fence_length_re.lastIndex = 0;
    while ((match = fence_length_re.exec(content)) !== null) {
        length = Math.max(length, match[1].length + 1);
    }
    return '`'.repeat(length);
};
```

Looks good otherwise.

Reply:
Fixed, thanks @andersk.